sdk/azidentity/chained_token_credential_test.go

```diff
@@ -113,6 +113,95 @@ func TestChainedTokenCredential_GetTokenFail(t *testing.T) {
 	}
 }
 
+func TestChainedTokenCredential_MultipleCredentialsGetTokenUnavailable(t *testing.T) {
+	credential1 := &TestCredential{responses: []testCredentialResponse{
+		{err: newCredentialUnavailableError("unavailableCredential1", "Unavailable expected error")},
+	}}
+	credential2 := &TestCredential{responses: []testCredentialResponse{
+		{err: newCredentialUnavailableError("unavailableCredential2", "Unavailable expected error")},
+	}}
+	credential3 := &TestCredential{responses: []testCredentialResponse{
+		{err: newCredentialUnavailableError("unavailableCredential3", "Unavailable expected error")},
+	}}
+	cred, err := NewChainedTokenCredential([]azcore.TokenCredential{credential1, credential2, credential3}, nil)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	_, err = cred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{liveTestScope}})
+	if err == nil {
+		t.Fatalf("Expected an error but did not receive one")
+	}
+	var authErr credentialUnavailableError
+	if !errors.As(err, &authErr) {
+		t.Fatalf("Expected CredentialUnavailableError, received %T", err)
+	}
+	expectedError := `ChainedTokenCredential: 
```

Comment:
Did you mean to change the message in this case? This test still passes, without an update

Reply:
I’ll update this in a bit

Reply:
So, the error received here is specifically the Credential Unavailable Error made if none of the sources were available. Credential unavailable errors don’t tend to have these introductory messages, as far as I can tell. If I provide a message, should it be `Authentication failed for…`? given that is typically seen related to the Authentication Failed Error and not the Credential Unavailable Error.

Reply:
> Credential unavailable errors don’t tend to have these introductory messages, as far as I can tell

They may not--I haven't reviewed them all yet. I think they should. Just something short to indicate the nature of the error. Remember that users get these errors from client methods. The most common error handling will be like:
```go
thing, err := client.Frobulate()
if err != nil {
    // do something with err.Error()
}
```
... so azidentity error messages should clearly indicate that authentication failed.

For this particular message, I don't think we need to distinguish authentication failed vs. credential unavailable. I think the important bits are
1. name of the credential that produced the error
2. nature of the error: can't get a token (exact failure mode isn't important)
3. error messages from the chain

For example, the message in Python works for me; it would be like
```
DefaultAzureCredential failed to acquire a token. Attempted credentials:
    ManagedIdentityCredential: ...
```

Reply:
(🤣 Frobulate 🥲)

Reply:
So, the credential unavailable error forces the `colon` right before the message. Without altering the credential unavailable error’s source, this is the closest I think I can get to the message you’re showing: https://github.com/Azure/azure-sdk-for-go/pull/16844/commits/f76915f7034782283c28d5434dea899fb264d75a

Let me know if you were expecting something different!